### PR TITLE
[DSS-340] add table of contents sidebar

### PR DESCRIPTION
**Pull request by QuintonJason**
# Description
This PR adds a table of contents as a right sidebar to our Storybook implementation. H2's in the documentation will also exist in the table of contents. 

|  Before  |  After  |
|--------|--------|
|Screen_Shot_2023-02-21_at_7_12_53_PM|Screen_Shot_2023-02-24_at_8_30_45_AM|

**Dev Dependencies:**
- `storybook: "^6.5.13"` to resolve TOC not displaying anchor links in the nav
- `storybook-docs-toc`
    - Depends on `styled-components`
- `@babel/plugin-syntax-jsx`

Fixes #(issue)
[DSS-340](https://kajabi.atlassian.net/browse/DSS-340)

## Type of change
- [ ] This change requires a documentation update

# How Has This Been Tested?
Docs only

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes

[DSS-340]: https://kajabi.atlassian.net/browse/DSS-340?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by QuintonJason** [comment]
Closing in favor of https://github.com/Kajabi/sage/pull/26

**Comment by QuintonJason** [comment]
Reopened because the dark mode has complexities that will prolong this release. No need to have it blocked

File: libs/core/package.json [hunk 1]
```diff
@@ -57,17 +57,19 @@
...
-    "@storybook/builder-webpack5": "^6.5.12",
-    "@storybook/manager-webpack5": "^6.5.12",
-    "@storybook/web-components": "^6.5.12",
+    "@storybook/addon-a11y": "^6.5.13",
+    "@storybook/addon-actions": "^6.5.13",
+    "@storybook/addon-essentials": "^6.5.13",
+    "@storybook/addon-links": "^6.5.13",
+    "@storybook/builder-webpack5": "^6.5.13",
+    "@storybook/manager-webpack5": "^6.5.13",
+    "@storybook/web-components": "^6.5.13",
```
Comment (ju-Skinner) [inline libs/core/package.json:72]: I know it's a minor update, but do we know what triggered the dependency update?

Comment (QuintonJason) [reply libs/core/package.json:72]: The author state that this is needed as the anchor links no longer working makes the plugin unusable: https://github.com/storybookjs/storybook/issues/18395

**Review by anechol — APPROVED** [review]
Love this change! Is it possible to smooth scroll the navigation?

**Review by goodwinchris — APPROVED** [review]
Love this!

**Comment by QuintonJason** [comment]
Closing this PR as Storybook now is updated to Storybook 7

**Pull request closed without merging**